tests_integration/test_nativeapp.py

```diff
@@ -0,0 +1,142 @@
+import pytest
+import os
+import uuid
+from pathlib import Path
+from contextlib import contextmanager
+
+from snowcli.cli.project.util import generate_user_env
+from tests.project.fixtures import *
+from tests.testing_utils.fixtures import temp_dir
+from tests_integration.snowflake_connector import test_database, snowflake_session
+from tests_integration.test_utils import (
+    row_from_snowflake_session,
+    contains_row_with,
+)
+
+
+USER_NAME = f"user_{uuid.uuid4().hex}"
+TEST_ENV = generate_user_env(USER_NAME)
+
+
+@contextmanager
+def pushd(dir: Path):
+    cwd = os.getcwd()
+    os.chdir(dir)
+    try:
+        yield dir
+    finally:
+        os.chdir(cwd)
+
+
+@pytest.mark.integration
+def test_nativeapp_init_run_without_modifications(
+    runner,
+    snowflake_session,
+    temp_dir,
+):
+    project_name = "myapp"
+    result = runner.invoke_with_config(
+        ["app", "init", project_name],
+        env=TEST_ENV,
+    )
+    assert result.exit_code == 0
+
+    with pushd(Path(os.getcwd(), project_name)):
```

Comment:
@sfc-gh-turbaszek introduced project_file context manager https://github.com/Snowflake-Labs/snowcli/commit/3b59d1cb8f6b03ba5358d65ca21e7f2fc8623d44#diff-ad6b0ce118624337f6faf4e75af2aac6cc563a8632cbd95e699a04c1786f185dR208-R216

Logic is almost the same, but we will be consistent

Reply:
Do you mind if we refactor in a different ticket? This isn't the only place it's used (also in non-integration tests for definitions).

Reply:
ok